homeassistant/components/shelly/sensor.py

```diff
@@ -696,6 +728,10 @@ def __init__(
         super().__init__(coordinator, block, attribute, description)
 
         self._attr_native_unit_of_measurement = description.native_unit_of_measurement
+        self._attr_suggested_unit_of_measurement = (
+            description.suggested_unit_of_measurement
+        )
+        self._attr_suggested_display_precision = description.suggested_display_precision
```

Comment:
```suggestion
```

setting in the description should be enough

https://github.com/home-assistant/core/blob/1759f58fc110b6c3ad81d859d6e5716182ad29d2/homeassistant/components/sensor/__init__.py#L372
https://github.com/home-assistant/core/blob/1759f58fc110b6c3ad81d859d6e5716182ad29d2/homeassistant/components/sensor/__init__.py#L405